### DISCUSSION TITLE
Separate /var partition

### DISCUSSION THREAD

**Discussion by keerthi-varuman** [ltsp/ltsp#347] [Q&A] [answered]
We have installed LTSP on Debian 10 using the chrootless method, and when booting the client we get the following error message (see the attached screenshot, right after the overlayfs mount).

![2020-12-11](https://user-images.githubusercontent.com/75841376/101904635-27b5c580-3bdc-11eb-9eb2-0d86879c461c.jpg)

We are using a separate partition for the /var filesystem on the LTSP server, as below:
```
root@rose.blue.av:~# df -h
Filesystem               Type      Size  Used Avail Use% Mounted on
udev                     devtmpfs  3.9G     0  3.9G   0% /dev
tmpfs                    tmpfs     789M   18M  772M   3% /run
/dev/mapper/rose.av-root ext4       28G   18G  8.4G  68% /
tmpfs                    tmpfs     3.9G  100K  3.9G   1% /dev/shm
tmpfs                    tmpfs     5.0M  4.0K  5.0M   1% /run/lock
tmpfs                    tmpfs     3.9G     0  3.9G   0% /sys/fs/cgroup
/dev/mapper/rose.av-opt  ext4       28G  145M   26G   1% /opt
/dev/mapper/rose.av-var  ext4       19G  526M   17G   3% /var
/dev/sda1                vfat      234M  5.1M  229M   3% /boot/efi
/dev/mapper/rose.av-home ext4      2.6T  226M  2.4T   1% /home
tmpfs                    tmpfs     789M   16K  789M   1% /run/user/117
tmpfs                    tmpfs     789M     0  789M   0% /run/user/0
```

And we have realized that, apparently, the contents of the /var tree are not included in the LTSP image.
```
root@rose.blue.av:~# mount /srv/ltsp/images/x86_64.img /mnt/tmp/
root@rose.blue.av:~# ls -a /mnt/tmp/var/
.  ..
root@rose.blue.av:~# 
```

For the purpose of testing, we have tried the following, and then the client booted ok:
```
cd /srv/ltsp/images/
unsquashfs x86_64.img
rsync -avzh /var/ squashfs-root/var/
mksquashfs squashfs-root filesystem.squashfs
rm x86_64.img
mv filesystem.squashfs x86_64.img
```

**Comment by alkisg** [comment]
Hi, I converted this question into a discussion, as it's not a bug in the LTSP code base.
Please read the [documentation about advanced image sources](https://ltsp.org/man/ltsp-ipxe/#advanced-image-sources).
Specifically, you need this:

```shell
ltsp image /,,/var,subdir=var
```

If you had a separate boot partition instead, you'd need this:

```shell
ltsp image /,,/boot,subdir=boot
```

**Reply by keerthi-varuman** [reply]
Thank you, yes all works fine now. but I need to include /var and /opt at the same time because I have /var and /opt is a separate partition. how to do that?

**Comment by alkisg** [comment, marked as answer]
This should do it:

```shell
ltsp image /,,/var,subdir=var,,/opt,subdir=opt
```

**Reply by keerthi-varuman** [reply]
Working  thank you

**Comment by keerthi-varuman** [comment]
guide me how to mask services on LTSP client  like  (OpenVPN and some other services )

**Comment by alkisg** [comment]
Read about MASK_SYSTEM_SERVICES in the [ltsp.conf man page](https://ltsp.org/man/ltsp.conf/), and if you still have questions, open a new discussion here with an appropriate subject.

You may also mark this question as solved by clicking the check mark [v] in one of the answers.
